metrics/fork/fork_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fork
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/uber/jaeger-lib/metrics"
+	"github.com/uber/jaeger-lib/metrics/metricstest"
+)
+
+func TestAPICheck(t *testing.T) {
+	var v interface{} = &Factory{}
+	_, ok := v.(metrics.Factory)
+	assert.True(t, ok)
+}
```

Comment:
Much simpler with a compile-time check:

```suggestion
var _ metrics.Factory = (*Factory)(nil)
```